Console/Command/ResetYotpoSync.php

```diff
@@ -0,0 +1,225 @@
+                'All Entities'
+            ),
+            new InputOption(
+                self::STORE_ID,
+                null,
+                InputOption::VALUE_REQUIRED,
+                'Store ID'
+            ),
+        ];
+
+        $this->setName('yotpo:resetsync')
+            ->setDescription('Reset Sync')
+            ->setDefinition($options);
+
+        parent::configure();
+    }
+
+    /**
+     * @param InputInterface $input
+     * @param OutputInterface $output
+     * @return $this|int
+     * @throws LocalizedException
+     * @throws NoSuchEntityException
+     */
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $this->setAreaCode();
+
+        $isAllOption = $input->getOption(self::YOTPO_ENTITY_ALL);
+        $storeId = $input->getOption(self::STORE_ID);
+        if ($storeId) {
+            $storeIds = [$storeId];
+        } else {
+            $storeIds = $this->config->getAllStoreIds();
+        }
+        if (!$storeIds) {
+            return $this;
+        }
+        foreach ($storeIds as $storeId) {
+            if ($isAllOption) {
+                $this->resetAllSync($storeId);
+            } else {
+                $yotpoEntityInput = $input->getOption(self::YOTPO_ENTITY);
+                if (!$yotpoEntityInput) {
+                    return $this;
+                }
+                if (!is_array($yotpoEntityInput)) {
+                    $yotpoEntityInput = [$yotpoEntityInput];
+                }
+                $this->resetYotpoSync($yotpoEntityInput, $storeId, $output);
+            }
+        }
+
+        return $this;
+    }
+
+    /**
+     * @return void
+     * @throws LocalizedException
+     */
+    public function setAreaCode()
+    {
+        try {
+            $this->appState->getAreaCode();
+        } catch (LocalizedException $e) {
+            $this->appState->setAreaCode(\Magento\Framework\App\Area::AREA_CRONTAB);
+        }
+    }
+
+    /**
+     * @param int $storeId
+     * @return void
+     */
+    public function resetAllSync($storeId)
+    {
+        $this->syncReset->resetSync($storeId);
+    }
+
+    /**
+     * @param int $storeId
+     * @return void
+     */
+    public function resetCatalogSync($storeId)
+    {
+        $this->syncReset->resetCatalogSync($storeId);
+    }
+
+    /**
+     * @param int $storeId
+     * @return void
+     */
+    public function resetCustomersSync($storeId)
+    {
+        $this->syncReset->resetCustomersSync($storeId);
+    }
+
+    /**
+     * @param int $storeId
+     * @return void
+     */
+    public function resetOrdersSync($storeId)
+    {
+        $this->syncReset->resetOrdersSync($storeId);
+    }
+
+    /**
+     * @param array <bool|string|null> $yotpoEntityInput
+     * @param int $storeId
+     * @param OutputInterface $output
+     * @return void
+     * @throws NoSuchEntityException
+     */
+    protected function resetYotpoSync($yotpoEntityInput, $storeId, OutputInterface $output)
```

Comment:
Please change to `resetYotpoSyncByEntity`

Reply:
Understood. We will update this